### PR TITLE
fpgainfo: add support for revision 2 of port errors

### PR DESCRIPTION
**Pull request by fpgamatt**
With Intel OFS, the definition of the port error bits have
changed to reflect the transition from CCI-P to AXI-S TLP.
The fpgainfo command now checks the revision sysfs entry
to use the correct port error bit definitions.

Signed-off-by: Matthew Gerlach <matthew.gerlach@linux.intel.com>

### CODE REVIEW

File: tools/fpgainfo/errors.c [hunk 1]
```diff
@@ -316,6 +337,29 @@ fpga_result errors_filter(fpga_properties *filter, int argc, char *argv[])
...
+		OPAE_MSG("Failed to get token Object");
+		return res;
+	}
+
+	res = fpgaObjectRead64(fpga_object, value, 0);
+	if (res != FPGA_OK) {
+		OPAE_MSG("Failed to Read object ");
+	}
+
+	res = fpgaDestroyObject(&fpga_object);
```
Comment (r-rojo) [inline tools/fpgainfo/errors.c:356]: If line 351 fails, we lose the error code here.

File: tools/fpgainfo/errors.c [hunk 2]
```diff
@@ -400,8 +444,15 @@ static void print_errors_info(fpga_token token, fpga_properties props,
 			       error_value);
 
 			if (!strcmp(errinfos[i].name, "Errors")) {
-				size = PORT_ERROR_COUNT;
-				error_string = PORT_ERROR;
+				uint64_t revision = 0;
+				get_port_error_revision(token, &revision);
```
Comment (r-rojo) [inline tools/fpgainfo/errors.c:448]: Do we or should we bail here if we can't successfully determine the revision?

File: tools/fpgainfo/errors.c (hunk 1 above)
Comment (fpgamatt) [reply tools/fpgainfo/errors.c:356]: I will fix that cut n paste error.

File: tools/fpgainfo/errors.c (hunk 2 above)
Comment (fpgamatt) [reply tools/fpgainfo/errors.c:448]: My theory is that if the revision cannot be determined, then assume the original version.

I suppose another approach is that we don't decode at all if there is no revision.

Comment (r-rojo) [reply tools/fpgainfo/errors.c:448]: Either way (if we decode or not), I think we should print out an error/warning to stderr indicating that revision couldn't be determined.

File: tools/fpgainfo/errors.c [hunk 3]
```diff
@@ -316,6 +337,29 @@ fpga_result errors_filter(fpga_properties *filter, int argc, char *argv[])
 	return res;
 }
 
+static fpga_result get_port_error_revision(fpga_token token, uint64_t *value)
+{
+	fpga_result res = FPGA_OK;
+	fpga_object fpga_object;
+
+	res = fpgaTokenGetObject(token,"error*/revision" , &fpga_object, FPGA_OBJECT_GLOB);
```
Comment (tswhison) [inline tools/fpgainfo/errors.c:345]: s/token,"error*/revision" ,/token, "error*/revision",/

File: tools/fpgainfo/errors.c [hunk 4]
```diff
@@ -400,8 +445,27 @@ static void print_errors_info(fpga_token token, fpga_properties props,
 			       error_value);
 
 			if (!strcmp(errinfos[i].name, "Errors")) {
-				size = PORT_ERROR_COUNT;
-				error_string = PORT_ERROR;
+				uint64_t revision = 0;
+				res = get_port_error_revision(token, &revision);
+				if (res != FPGA_OK) {
+					printf("could not find port error revision - skipping decode\n");
```
Comment (r-rojo) [inline tools/fpgainfo/errors.c:451]: Should probably use OPAE_ERR macro here.

**Review by r-rojo — APPROVED** [review]

**Pull request merged**